R/r.R

```diff
@@ -179,62 +162,89 @@ rename_files <- function(old, new) {
 
 # helpers -----------------------------------------------------------------
 
-check_file_name <- function(name) {
-  if (!is_string(name)) {
-    ui_stop("Name must be a single string")
+compute_name <- function(name = NULL, ext = "R", error_call = caller_env()) {
+  if (!is.null(name)) {
+    check_file_name(name, call = error_call)
+
+    if (path_ext(name) == "") {
+      name <- path_ext_set(name, ext)
+    } else if (path_ext(name) != "R") {
+      cli::cli_abort(
+        "{.arg name} must have extension {.str {ext}}, not {.str {path_ext(name)}}.",
+        call = error_call
+      )
+    }
+    return(as.character(name))
   }
-  if (!valid_file_name(path_ext_remove(name))) {
-    ui_stop(c(
-      "{ui_value(name)} is not a valid file name. It should:",
-      "* Contain only ASCII letters, numbers, '-', and '_'."
-    ))
+
+  if (!rstudio_available()) {
+    cli::cli_abort(
+      "{.arg name} is absent but must be specified.",
+      call = error_call
+    )
   }
-  name
+  compute_active_name(
+    path = rstudioapi::getSourceEditorContext()$path,
+    ext = ext,
+    error_call = error_call
+  )
 }
 
-check_not_empty_file_name <- function(name = NULL) {
-  if (!is.null(name) && path_file(name) == "") {
-    ui_stop("Name must not be an empty string")
+compute_active_name <- function(path, ext, error_call = caller_env()) {
+  if (is.null(path)) {
+    cli::cli_abort(
+      c(
+        "No file is open in RStudio.",
+        i = "Please specify {.arg name}."
+      ),
+      call = error_call
+    )
   }
 
-  invisible(TRUE)
-}
+  ## rstudioapi can return a path like '~/path/to/file' where '~' means
+  ## R's notion of user's home directory
+  path <- proj_path_prep(path_expand_r(path))
 
-valid_file_name <- function(x) {
-  grepl("^[a-zA-Z0-9._-]+$", x)
-}
+  rel_path <- path_dir(proj_rel_path(path))
```

Comment:
`rel_path` feels like a confusing name for the immediate parent directory within the project, but I guess it's not terribly important.